zjit/src/hir.rs

```diff
@@ -4697,7 +4672,8 @@ impl Function {
                 self.assert_subtype(insn_id, index, types::Fixnum)
             }
             // Instructions with Hash operands
-            Insn::HashAref { hash, .. } => self.assert_subtype(insn_id, hash, types::Hash),
+            Insn::HashAref { hash, .. }
```

Comment:
looking at this ,we should probably do the same for aref: hash exact only, no subclasses. for now can you make this a separate case where the recv hash to be HashExact